internal/providers/pluginfw/resources/clusters/resource_library.go

```diff
@@ -0,0 +1,237 @@
+package clusters
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/terraform-provider-databricks/clusters"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/compute_tf"
+	"github.com/databricks/terraform-provider-databricks/libraries"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/databricks/databricks-sdk-go"
+)
+
+const libraryDefaultInstallationTimeout = 15 * time.Minute
+
+var _ resource.ResourceWithConfigure = &LibraryResource{}
+
+func ResourceLibrary() resource.Resource {
+	return &LibraryResource{}
+}
+
+func parseId(id string) (string, string) {
+	split := strings.SplitN(id, "/", 2)
+	if len(split) != 2 {
+		return "unknown", "unknown"
+	}
+	return split[0], split[1]
+}
+
+func readLibrary(ctx context.Context, w *databricks.WorkspaceClient, waitParams compute.Wait, libraryRep string, libraryExtended *LibraryExtended) diag.Diagnostics {
+	res, err := libraries.WaitForLibrariesInstalledSdk(ctx, w, waitParams, libraryDefaultInstallationTimeout)
+	if err != nil {
+		return diag.Diagnostics{diag.NewErrorDiagnostic("failed to wait for library installation", err.Error())}
+	}
+
+	for _, v := range res.LibraryStatuses {
+		thisRep := v.Library.String()
+		if thisRep == libraryRep {
+
+			diags := converters.GoSdkToTfSdkStruct(ctx, v.Library, libraryExtended)
+
+			if diags.HasError() {
+				return diags
+			}
+
+			libraryExtended.ClusterId = types.StringValue(waitParams.ClusterID)
```

Comment:
Noting that we're porting this resource, so this preserves the original behavior, but a question: do we actually need to do this? There are no computed fields in this resource. In theory, we don't even need to update the state at the end of the `Create()` or `Read()` methods.